### PR TITLE
Use latest builder image with yarn properly setup

### PR DESCRIPTION
**Pull request by halkeye**
See https://github.com/jenkins-infra/helpdesk/issues/4
and https://github.com/jenkins-infra/docker-builder/pull/34

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -84,16 +81,6 @@ spec:
...
-      steps {
-        sh '''
-          mkdir "${HOME}/.npm-packages"
-          npm config set prefix "${HOME}/.npm-packages"
-          npm install --global yarn
-        '''
-      }
-    }
-
     stage('NPM Install') {
```
Comment (timja) [inline Jenkinsfile_k8s]: nit outdated from other PRs where it switched to yarn

**Review by timja — APPROVED** [review]

File: renovate.json [hunk 2]
```diff
@@ -26,6 +26,12 @@
         }
     ],
     "packageRules": [
+        {
+            "matchPackageNames": [
+                "jenkinsciinfra/builder"
```
Comment (timja) [inline renovate.json:31]: is renovate going to be able to update that? we'll probably remove the builder from the repo at some point anyway and make it the default for infra.ci so it doesn't need to be here

File: Jenkinsfile_k8s (hunk 1 above)
Comment (halkeye) [reply Jenkinsfile_k8s]: yea thats fair, it was kinda intentional. I felt NPM install was clearer than yarn, but I've switched it to "install dependencies"

File: renovate.json (hunk 2 above)
Comment (halkeye) [reply renovate.json:31]: it *Should* based on https://github.com/jenkins-infra/plugin-site/blob/master/renovate.json#L14-L27 but i haven't yet been able to figure out how to get it to test the configs.

Comment (timja) [reply renovate.json:31]: o that's cool

**Review by timja — APPROVED** [review]

**Pull request merged**